sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/StreamObserverFactory.java

```diff
@@ -16,57 +16,57 @@
  * limitations under the License.
  */
 
-package org.apache.beam.fn.harness.stream;
+package org.apache.beam.sdk.fn.stream;
 
 import io.grpc.stub.CallStreamObserver;
 import io.grpc.stub.StreamObserver;
-import java.util.List;
 import java.util.concurrent.ExecutorService;
-import java.util.function.Function;
-import org.apache.beam.sdk.extensions.gcp.options.GcsOptions;
-import org.apache.beam.sdk.fn.stream.AdvancingPhaser;
-import org.apache.beam.sdk.fn.stream.BufferingStreamObserver;
-import org.apache.beam.sdk.fn.stream.DirectStreamObserver;
-import org.apache.beam.sdk.fn.stream.ForwardingClientResponseObserver;
-import org.apache.beam.sdk.options.ExperimentalOptions;
-import org.apache.beam.sdk.options.PipelineOptions;
 
 /**
- * Uses {@link PipelineOptions} to configure which underlying {@link StreamObserver} implementation
- * to use.
+ * Creates factories which determine an underlying {@link StreamObserver} implementation
+ * to use in to interact with fn execution APIs.
  */
 public abstract class StreamObserverFactory {
-  public static StreamObserverFactory fromOptions(PipelineOptions options) {
-    List<String> experiments = options.as(ExperimentalOptions.class).getExperiments();
-    if (experiments != null && experiments.contains("beam_fn_api_buffered_stream")) {
-      int bufferSize = Buffered.DEFAULT_BUFFER_SIZE;
-      for (String experiment : experiments) {
-        if (experiment.startsWith("beam_fn_api_buffered_stream_buffer_size=")) {
-          bufferSize = Integer.parseInt(
-              experiment.substring("beam_fn_api_buffered_stream_buffer_size=".length()));
-        }
-      }
-      return new Buffered(options.as(GcsOptions.class).getExecutorService(), bufferSize);
-    }
+  /**
+   * Create a buffering {@link StreamObserverFactory} with the specified {@link ExecutorService} and
+   * the default buffer size.
+   */
+  public static StreamObserverFactory createBuffered(ExecutorService executorService) {
```

Comment:
<!--new_thread; commit:a45dd983c9cc24cad68f32ea09a82f93d7d5f3af; resolved:1-->
Nit: no need for `create` on method names if the object is immutable - does the caller of this method need to know if this is a fresh object or not? If not, just `buffered(...)` since any creation is an implementation detail.